ios/app/brave_core_main.mm

```diff
@@ -95,9 +109,30 @@ - (instancetype)initWithUserAgent:(NSString*)userAgent
     web::SetWebClient(_webClient.get());
 
     _delegate.reset(new BraveMainDelegate());
-    _delegate->SetSyncServiceURL(base::SysNSStringToUTF8(syncServiceURL));
 
     web::WebMainParams params(_delegate.get());
+    NSMutableArray* arguments =
+        [[[NSProcessInfo processInfo] arguments] mutableCopy];
+    NSMutableArray* switches = [[NSMutableArray alloc] init];
+    for (NSString* key in additionalSwitches) {
+      if (![additionalSwitches[key] isKindOfClass:NSString.class]) {
+        continue;
+      }
+      [switches
+          addObject:[NSString stringWithFormat:@"--%@=%@", key,
+                                               static_cast<NSString*>(
+                                                   additionalSwitches[key])]];
+    }
+    [arguments addObjectsFromArray:switches];
+    params.argc = [arguments count];
+    const char* argv[params.argc];
```

Comment:
This is illegal in `C++03, 11, 17, 20, 23` (all versions of C++) but legal in `C99`.
The reason it compiles is because it is a `clang` and `gcc` extension, but it can blow out the stack if too many args are passed.
Also, passing this to `params.argv` is passing a `stack-allocated` variable.

We should have a `std::unique_ptr<char[]> argv;` stored somewhere.
Then `params.argv = this->argv.get()`
 and inside of `BraveCoreMain::~BraveCoreMain()` destructor, we call `argv.reset()`

Reply:
This is basically copy-paste of `IOSChromeMain`: https://source.chromium.org/chromium/chromium/src/+/main:ios/chrome/app/startup/ios_chrome_main.mm;l=29

Reply:
Allowing this to go through because **Chromium** is already doing it *sigh*: https://source.chromium.org/chromium/chromium/src/+/main:ios/chrome/app/startup/ios_chrome_main.mm;l=30?q=NSProcessInfo&ss=chromium%2Fchromium%2Fsrc:ios%2F

The lifetime of `params` is bound to the end of the function as well. So we can ignore this.